numba_cuda/numba/cuda/cudadrv/driver.py

```diff
@@ -2704,6 +2710,13 @@ def add_file_guess_ext(self, path_or_code):
                         "Don't know how to link file with extension "
                         f"{ext}"
                     )
+                if ignore_nonlto and kind != FILE_EXTENSION_MAP["ltoir"]:
```

Comment:
There's a nuance here where fatbin object can also only contains LTOIR. Though that requires building additional bindings for `cuobjdump` and outputting the result.

Reply:
In [bac11f6](https://github.com/NVIDIA/numba-cuda/pull/62/commits/bac11f68a9be548f175abfe9e7698d78eb8c6b7e) I made a new created a new fatbin object that contains both LTOIR and SASS and feed into the linker for testing. I added a helper function that invokes `cuobjdump` to extract object types from the fatbin / object. We allow and fatbin / object that contains LTOIR to be passed into the linker.